cpu/rpx0xx/periph/usbdev.c

```diff
@@ -0,0 +1,746 @@
+    
+    if (status & USBCTRL_REGS_INTS_DEV_SOF_Msk) {
+        /* Start of frame received */
+        uint32_t frame_number = USBCTRL_REGS->SOF_RD;
+        /* we have to read the frame number to reset the interrupt,
+           but do not need it */
+        (void) frame_number;
+        _hw_usb_dev.dev.cb(dev, USBDEV_EVENT_SOF);
+    }
+    if (status & USBCTRL_REGS_INTS_BUS_RESET_Msk) {
+        /* Bus reset received */
+        _hw_usb_dev.dev.cb(dev, USBDEV_EVENT_RESET);
+        io_reg_atomic_clear(&USBCTRL_REGS->SIE_STATUS,
+                            USBCTRL_REGS_SIE_STATUS_BUS_RESET_Msk);
+    }
+    if (status & USBCTRL_REGS_INTS_DEV_SUSPEND_Msk) {
+        /* Bus suspend state has changed */
+        if (_hw_usb_dev.suspended) {
+            _hw_usb_dev.suspended = false;
+            _hw_usb_dev.dev.cb(dev, USBDEV_EVENT_RESUME);
+            /* TODO: pm_block() for MODULE_PM_LAYERED */
+        } else {
+            _hw_usb_dev.suspended = true;
+            _hw_usb_dev.dev.cb(dev, USBDEV_EVENT_SUSPEND);
+            /* TODO: pm_unblock() for MODULE_PM_LAYERED */
+        }
+        io_reg_atomic_clear(&USBCTRL_REGS->SIE_STATUS,
+                            USBCTRL_REGS_SIE_STATUS_SUSPENDED_Msk);
+    }
+    if (status & USBCTRL_REGS_INTS_DEV_CONN_DIS_Msk) {
+        /* Host connected or disconnected */
+        if (_hw_usb_dev.connected) {
+            _hw_usb_dev.connected = false;
+            _hw_usb_dev.dev.cb(dev, USBDEV_EVENT_HOST_DISCONNECT);
+        } else {
+            _hw_usb_dev.connected = true;
+            _hw_usb_dev.dev.cb(dev, USBDEV_EVENT_HOST_CONNECT);
+        }
+        io_reg_atomic_clear(&USBCTRL_REGS->SIE_STATUS,
+                            USBCTRL_REGS_SIE_STATUS_CONNECTED_Msk);
+    }
+    /* all interrupts should be now cleared */
+    status = USBCTRL_REGS->INTS;
+    if (status) {
+        DEBUG("[rpx0xx usb] Unhandled interrupt, INTS=%lx", status);
+    }
+    _enable_irq();
+}
+
+static void _usbdev_ep_esr(usbdev_ep_t *ep)
+{
+    if (_reg_get_flag_for_ep(&USBCTRL_REGS->BUFF_STATUS, ep)) {
+        _reg_clear_flag_for_ep(&USBCTRL_REGS->BUFF_STATUS, ep);
+        ep->dev->epcb(ep, USBDEV_EVENT_TR_COMPLETE);
+    }
+    if (_reg_get_flag_for_ep(&USBCTRL_REGS->EP_STATUS_STALL_NAK, ep)) {
+        _reg_clear_flag_for_ep(&USBCTRL_REGS->EP_STATUS_STALL_NAK, ep);
+        ep->dev->epcb(ep, USBDEV_EVENT_TR_STALL);
+    }
+    _enable_ep_irq(ep);
+}
+
+void isr_usbctrl(void)
+{
+    if (USBCTRL_REGS->BUFF_STATUS) {
+        /* Endpoint specific interrupt */
+        for (uint8_t i = 0; i < USBDEV_NUM_ENDPOINTS * 2; i++) {
+            rpx0xx_usb_ep_t* hw_ep = _hw_usb_dev.hw_ep;
+            if (_reg_get_flag_for_ep(&USBCTRL_REGS->BUFF_STATUS, &hw_ep->ep)) {
+                _disable_ep_irq(&hw_ep->ep);
+                _hw_usb_dev.dev.epcb(&hw_ep->ep, USBDEV_EVENT_ESR);
+            }
+        }
+    }
+    if (USBCTRL_REGS->INTS) {
+        /* Device specific interrupt */
+        _disable_irq();
```

Comment:
Disabling all IRQs here will have the side effect to also clear INTS register.
So when USBUS call its ESR callback, it doesn't process any IRQs as INTS is 0.
Depending on the interruption's source, you may need to clear this register either here or in `usbdev_esr`